### ISSUE TITLE
Build failure for `libp2p-noise` crate

### ISSUE THREAD

**Issue by emmyoh** [libp2p/rust-libp2p#2582]
Error:
```rust
error[E0282]: type annotations needed
   --> /home/emil/.cargo/registry/src/github.com-1ecc6299db9ec823/libp2p-noise-0.32.0/src/protocol/x25519.rs:221:45
    |
221 |         curve25519_sk.copy_from_slice(&hash.as_ref()[..32]);
    |                                        -----^^^^^^--
    |                                        |    |
    |                                        |    cannot infer type for type parameter `T` declared on the trait `AsRef`
    |                                        this method call resolves to `&T`
    |
    = note: type must be known at this point

For more information about this error, try `rustc --explain E0282`.
error: could not compile `libp2p-noise` due to previous error
```

Rust toolchain: `nightly-x86_64-unknown-linux-gnu`
Cargo version: `cargo 1.61.0-nightly (65c8266 2022-03-09)`
Rust compiler version: `rustc 1.61.0-nightly (1bfe40d11 2022-03-18)`

**Comment by mxinden** [comment]
Which version of the `libp2p-noise` crate are you using?

Note that this code has changed last year:

https://github.com/libp2p/rust-libp2p/blob/0c24821a144b10ba805e3648e05c5d44bbd59e42/transports/noise/src/protocol/x25519.rs#L215-L226

**Comment by emmyoh** [comment]
> Which version of the `libp2p-noise` crate are you using?
> 
> Note that this code has changed last year:
> 
> https://github.com/libp2p/rust-libp2p/blob/0c24821a144b10ba805e3648e05c5d44bbd59e42/transports/noise/src/protocol/x25519.rs#L215-L226

Running `cargo tree` on my project shows that I'm using `libp2p-noise v0.32.0`, which is the latest published release on crates.io.

**Comment by tdelabro** [comment]
We got the same issue. @emmyoh did you fixed it ?

**Comment by emmyoh** [comment]
> We got the same issue. @emmyoh did you fixed it ?

Unfortunately, @tdelabro, I could not find a solution. Would appreciate if a maintainer/contributor of this crate could help (attempting replication, debugging, assigning someone, etc?)

**Comment by elenaf9** [comment]
The latest released version of `libp2p-noise` is `v0.35.0`: https://crates.io/crates/libp2p-noise/0.35.0. As @mxinden pointed out, the code that is not compiling changed last year in #2264, which was part of release `v0.33.0`. So updating to a version `>= 0.33.0` should fix your issue.

**Comment by emmyoh** [comment]
Hi @elenaf9, you appear to be correct. I'm not sure where I saw that `v0.32.0` was the latest version, but clearly it's not. Closing.
